parser/yaml/yaml.go

```diff
@@ -29,12 +29,35 @@ func (yp *Parser) Unmarshal(p []byte, v interface{}) error {
 }
 
 func separateSubDocuments(data []byte) [][]byte {
+	// Determine line ending style
 	linebreak := "\n"
-	if bytes.Contains(data, []byte("\r\n---\r\n")) {
+	if bytes.Contains(data, []byte("\r\n")) {
 		linebreak = "\r\n"
 	}
+	separator := fmt.Sprintf("%s---%s", linebreak, linebreak)
```

Comment:
Since we're working with bytes here already, I'd probably just stick to that. 

```go
var (
	lf   = []byte{'\n'}
	crlf = []byte{'\r', '\n'}
	sep  = []byte{'-', '-', '-'}
)

func separateSubDocuments(data []byte) [][]byte {
	// Determine line ending style
	linebreak := lf
	if bytes.Contains(data, crlf) {
		linebreak = crlf
	}

	separator := slices.Concat(linebreak, sep, linebreak)

	// Count actual document separators
	parts := bytes.Split(data, separator)
```

But that's optional :)

Reply:
Good idea! I'll do this